packages/core/src/browser/preferences/preference-proxy.ts

```diff
@@ -17,29 +17,37 @@
 // tslint:disable:no-any
 
 import { Disposable, DisposableCollection, Event, Emitter } from '../../common';
-import { PreferenceService, PreferenceChange } from './preference-service';
+import { PreferenceService, PreferenceDataChange } from './preference-service';
 import { PreferenceSchema } from './preference-contribution';
 
 export interface PreferenceChangeEvent<T> {
-    readonly preferenceName: keyof T
-    readonly newValue?: T[keyof T]
-    readonly oldValue?: T[keyof T]
+    readonly preferenceName: keyof T;
+    readonly newValue?: T[keyof T];
+    readonly oldValue?: T[keyof T];
+    canAffect(resourceUri?: string): boolean;
```

Comment:
why `can`? Should not it tell whether it is affected or not? just `isAffected` or `affects`?